core/src/main/java/io/kestra/core/lock/LockService.java

```diff
@@ -0,0 +1,160 @@
+package io.kestra.core.lock;
+
+import io.kestra.core.repositories.LockRepositoryInterface;
+import io.kestra.core.server.ServerInstance;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+/**
+ * This service provides facility for executing Runnable and Callable tasks inside a lock.
+ * Note: it may be handy to provide a tryLock facility that, if locked, skip executing the Runnable or Callable and exit immediately.
+ *
+ * @implNote There are no expiry for locks so a service may hold a lock infinitely until the service is restarted as the
+ *           liveness mechanism release all locks when the service is unreachable.
+ *           This may be improved at some point by adding an expiry (for ex 30s) and running a thread that will periodically
+ *           increase the expiry for all exiting locks. This should allow quicker recovery of zombie locks than relying on the liveness mechanism,
+ *           as a service wanted to lock an expired lock would be able to take it over.
+ */
+@Slf4j
+@Singleton
+public class LockService {
```

Comment:
I prefer to have a `isLocked` method exposed by this service than having to do `!lockRepositoryInterface.findById(...).isEmpty()`. As a user I don't want to know that the locked mecanism is based on a repository. Also, it's painful to have to inject the repository just to check if something is locked while only  the LockService is needed. This will reduce the coupling with the `LockRepositoryInterface`.